module/VuFind/src/VuFind/Search/EDS/AbstractEDSParams.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Common EDS & EPF API Params
+ *
+ * PHP version 8
+ *
+ * Copyright (C) EBSCO Industries 2013
+ * Copyright (C) The National Library of Finland 2022
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  EBSCO
+ * @author   Michelle Milton <mmilton@epnet.com>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Search\EDS;
+
+use VuFindSearch\ParamBag;
+
+/**
+ * Common EDS & EPF API Params
+ *
+ * @category VuFind
+ * @package  EBSCO
+ * @author   Michelle Milton <mmilton@epnet.com>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class AbstractEDSParams extends \VuFind\Search\Base\Params
+{
+    /**
+     * Set up filters based on VuFind settings.
+     *
+     * @param ParamBag $params Parameter collection to update
+     *
+     * @return void
+     */
+    public function createBackendFilterParameters(ParamBag $params)
```

Comment:
> If it would help, I can do some investigation and open a PR to address the existing EDS issues, and then we can just adjust this one to match.

That would certainly help.  I'm constantly trying to make sense of the EDS API as part of this, so it helps if EDS.ini is as close to correct as possible.

> I notice that you've removed the second Options parameter here... which makes sense, since it appears to be unused, and the Options object could easily be obtained through $this->getOptions() from inside the method if needed anyway. However, this is a minor BC break, so there's a question of whether it's worth retaining the parameter in case somebody relies on it in a local subclass, or if we should just drop it and put a note in the changelog. I think the latter is probably fine, but I'm wracking my brain to figure out why this was here in the first place, in case there's a valid use case I'm somehow forgetting about.

Yeah, I removed the parameter because when I call it from the new EPF/Params.php, I didn't have any $options to pass in, and then I noticed that the method ignored the parameter in any case.  I wrote up the code to keep the parameter but set the default to null, can commit that if you like, if you prefer keeping it for BC.

Reply:
Here's a PR to update EDS.ini: https://github.com/vufind-org/vufind/pull/2950 -- I double-checked the code to be sure that these changes are accurate/safe. I'm honestly not sure what organization_id is used for (in our local configuration, the value is set to a blank string), but it does get passed along to EDS in the code, and the value in EDS.ini is obviously a fake example, so changing it seems entirely harmless.

Regarding the second parameter to createBackendFilterParameters, I'll just add a TODO note to indicate the possible minor BC break. I can't think of a good reason to keep the parameter since the options object is accessible by other means anyway. Maybe there's some bizarre edge case where you would want to pass a _different_ Options object instead of the one already inside the Params object, but I can't think of a good reason for that -- and there would be other ways to achieve the same effect anyway.